community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala

```diff
@@ -216,9 +216,15 @@ abstract class CompiledExecutionResult(taskCloser: TaskCloser, statement:Stateme
   }
 
   private def makeValueJavaCompatible(value: Any): Any = value match {
-    case iter: Seq[_]    => iter.map(makeValueJavaCompatible).asJava
-    case iter: Map[_, _] => Eagerly.immutableMapValues(iter, makeValueJavaCompatible).asJava
-    case x               => x
+    case seq: Seq[_] => seq.map(makeValueJavaCompatible).asJava
+    case map: Map[_, _] => Eagerly.immutableMapValues(map, makeValueJavaCompatible).asJava
+    case x => x
+  }
+
+  private def makeValueScalaCompatible(value: Any): Any = value match {
+    case list: java.util.List[_] => list.asScala.map(makeValueScalaCompatible).toList
+    case map: java.util.Map[_, _] => map.asScala.map { case (k, v) => k -> makeValueScalaCompatible(v) }.toMap
```

Comment:
I think you want to do `Eagerly.immutableMapValues(map.asScala, makeValueScalaCompatible)` Not sure if it matters in this case but it is at least constistent